augur/tasks/git/util/facade_worker/facade_worker/utilitymethods.py

```diff
@@ -205,3 +215,55 @@ def update_facade_scheduling_fields(session, repo_git, weight, commit_count):
 
 	session.execute(update_query)
 	session.commit()
+
+def facade_bulk_insert_commits(session,records):
```

Comment:
I think this method might be cleaner if it took in the list of records and then tried to insert them. And if an exception occurs and the insert was on more than one row then we log the multiple record error message, and split in half then try again. But if an exception occurs and the number or records is 1 then we check if it a DataError exception and do all the logic and error logging for a single record insert. Put simply it moves the if len(records) == 1 into the exception block to determine what to log or how to react

Reply:
I implemented this change. I think it makes it slightly more readable